File: December - 31/KishoreMuruganantham_sudoku_solver.R
input <- file("stdin", "r")
lines <- readLines(input)

board <- matrix("", nrow = 9, ncol = 9)
for (i in 1:9) {
  row_vals <- unlist(strsplit(lines[i], "\\s+"))
  board[i, ] <- row_vals
}

is_valid <- function(r, c, k) {
  if (any(board[r, ] == k)) return(FALSE)
  if (any(board[, c] == k)) return(FALSE)
  
  r_start <- floor((r - 1) / 3) * 3 + 1
  c_start <- floor((c - 1) / 3) * 3 + 1
  
  sub_box <- board[r_start:(r_start + 2), c_start:(c_start + 2)]
  if (any(sub_box == k)) return(FALSE)
  
  return(TRUE)
}

solve <- function() {
  for (i in 1:9) {
    for (j in 1:9) {
      if (board[i, j] == ".") {
        for (k in 1:9) {
          sgs <- as.character(k)
          if (is_valid(i, j, sgs)) {
            board[i, j] <<- sgs
            if (solve()) return(TRUE)
            board[i, j] <<- "."
          }
        }
        return(FALSE)
      }
    }
  }
  return(TRUE)
}

if (solve()) {
  for (i in 1:9) {
    cat(paste(board[i, ], collapse = " "), "\n", sep = "")
  }
}

File: December - 31/KishoreMuruganantham_sudoku_solver.swift
import Foundation
var lines = [String]()
while let line = readLine() { lines.append(line) }
if lines.count >= 9 {
    var board = [[String]]()
    for i in 0..<9 {
        let row = lines[i].split(separator: " ").compactMap { String($0) }
        board.append(row)
    }
    
    func isValid(_ r: Int, _ c: Int, _ k: String) -> Bool {
        for j in 0..<9 {
            if board[r][j] == k { return false }
        }
        for i in 0..<9 {
            if board[i][c] == k { return false }
        }
        let rStart = (r / 3) * 3
        let cStart = (c / 3) * 3
        for i in 0..<3 {
            for j in 0..<3 {
                if board[rStart + i][cStart + j] == k { return false }
            }
        }
        return true
    }
    
    func solve() -> Bool {
        for i in 0..<9 {
            for j in 0..<9 {
                if board[i][j] == "." {
                    for n in 1...9 {
                        let s = String(n)
                        if isValid(i, j, s) {
                            board[i][j] = s
                            if solve() { return true }
                            board[i][j] = "."
                        }
                    }
                    return false
                }
            }
        }
        return true
    }
    
    if solve() {
        for row in board {
            print(row.joined(separator: " "))
        }
    }
}
